Add wordpress admin theme

KevinBatdorf opened the pull request:
Let me know what you think. WordPress doesn't have modals so I basically just left it as the dark theme on modals and focused on the toast messages. if they ever add modals I'll come back and fix it :) 

I tried to make it look similar whether using the built-in Sweetalert2 icons or the built-in WordPress icons.

One thing I noticed that might be good to add is a class to the container that tells you which icon is active. Then for example, the timer bar color could reflect the current state. Not important though for this.

Some examples using the WP dashicon set against the WordPress admin panel:
<img width="680" alt="Screen Shot 2019-12-09 at 11 38 35 PM" src="https://user-images.githubusercontent.com/1478421/70455125-51767100-1ade-11ea-9458-72b1e748ea74.png">
<img width="680" alt="Screen Shot 2019-12-09 at 11 39 57 PM" src="https://user-images.githubusercontent.com/1478421/70455126-520f0780-1ade-11ea-9280-184f6f040914.png">
<img width="681" alt="Screen Shot 2019-12-09 at 11 41 20 PM" src="https://user-images.githubusercontent.com/1478421/70455127-520f0780-1ade-11ea-8852-1fb7318d4c42.png">
<img width="682" alt="Screen Shot 2019-12-09 at 11 39 30 PM" src="https://user-images.githubusercontent.com/1478421/70455131-52a79e00-1ade-11ea-9194-2ecde283f42a.png">

Modified sweetalert icon (border removed):
<img width="680" alt="Screen Shot 2019-12-09 at 11 42 48 PM" src="https://user-images.githubusercontent.com/1478421/70455133-52a79e00-1ade-11ea-9cb7-34d5845a920a.png">

Let me know if you want anything tweaked, updated, etc.
 
Closes https://github.com/sweetalert2/sweetalert2-themes/issues/39

limonte commented:
Screenshots look sexy! :heart_eyes: Outstanding work @KevinBatdorf THANK YOU!

> One thing I noticed that might be good to add is a class to the container that tells you which icon is active. Then for example, the timer bar color could reflect the current state. Not important though for this.

Sounds like a valid feature-request. Reported: https://github.com/sweetalert2/sweetalert2/issues/1842
I'll implement it right away, it's trivial to add.

Would you mind resolving the conflict in `index.html` after merging #46?

KevinBatdorf commented:
@limonte I resolved the conflict. I also added a check during the watch function that will make sure the theme folder is available. I kept typing "wordpress" instead of "wordpress-admin" so this fixed that =/

I also updated the packages here and saw the icon fix you added, thanks!

<img width="653" alt="Screen Shot 2019-12-10 at 8 35 02 AM" src="https://user-images.githubusercontent.com/1478421/70487777-12b9d880-1b29-11ea-9476-ca231b5c7669.png">

KevinBatdorf commented:
You're probably offline at the moment, but if you could hold off on merging this for now I want to update the modals a bit to differentiate it more from the dark theme

KevinBatdorf commented:
Okay I added some modal styles to make it more of a unique theme. Ready to go when you are! Let me know if you want anything changed.

Here's some screenshots using SweetAlert2 defaults. 

<img width="642" alt="Screen Shot 2019-12-10 at 10 31 51 AM" src="https://user-images.githubusercontent.com/1478421/70493378-0db15500-1b3a-11ea-9ec1-e9497bc74e5b.png">
<img width="652" alt="Screen Shot 2019-12-10 at 10 31 36 AM" src="https://user-images.githubusercontent.com/1478421/70493379-0db15500-1b3a-11ea-833b-b0d9a2b59636.png">
<img width="651" alt="Screen Shot 2019-12-10 at 10 31 29 AM" src="https://user-images.githubusercontent.com/1478421/70493380-0db15500-1b3a-11ea-8ad8-f181fd013aec.png">
<img width="684" alt="Screen Shot 2019-12-10 at 10 31 21 AM" src="https://user-images.githubusercontent.com/1478421/70493381-0e49eb80-1b3a-11ea-9c95-7644f47f13b0.png">


And some that I tested in my WP application:

<img width="623" alt="Screen Shot 2019-12-10 at 10 42 43 AM" src="https://user-images.githubusercontent.com/1478421/70493411-24f04280-1b3a-11ea-9bc2-8e079364730b.png">
<img width="625" alt="Screen Shot 2019-12-10 at 10 42 53 AM" src="https://user-images.githubusercontent.com/1478421/70493412-24f04280-1b3a-11ea-8671-77feaa48db44.png">
<img width="592" alt="Screen Shot 2019-12-10 at 10 43 10 AM" src="https://user-images.githubusercontent.com/1478421/70493413-24f04280-1b3a-11ea-97c7-6a3e2d7ee9f6.png">

limonte commented:
OUTSTANDING WORK!

Totally love these toasts! :heart_eyes: :heart_eyes: :heart_eyes: 

I'll review your code during this day!

KevinBatdorf commented:
Thanks @limonte . Looking at it again now I might need to make a couple more adjustments, so I might be pushing a few more minor commits

KevinBatdorf commented:
Okay maybe you can take a look and see what needs to be improved on. The difficulty seems to be in resizing the icons. I used a scale transform as a shortcut as adjusting the width of the icon directly would require fine-tuning each of the the individual pieces as well. Unless I'm overlooking something.

I'll try to test it in other browsers tomorrow too. Signing off shortly. Thanks!

limonte commented on wordpress-admin/wordpress-admin.scss:
```diff
@@ -0,0 +1,192 @@
+
+// Overrides
+.swal2-container {
+  padding: 24px 16px;
```
use `$swal2-container-padding`

limonte replied:
use `$swal2-container-padding`

limonte commented on wordpress-admin/wordpress-admin.scss:
```diff
@@ -0,0 +1,192 @@
+// Overrides
+.swal2-container {
+  padding: 24px 16px;
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
```
we prefer not setting `font-family` for popups, as they should be aligned with an app, e.g. inherit `font-family`. I think the app `font-family` for Wordpress Admin is `-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;` If so, this line isn't needed.

KevinBatdorf replied:
@limonte That's right that SweetAlert will inherit the font family while in the WP dashboard but if someone wants the WP style toasts in a non-WP application then it wont look like WordPress anymore unless they set the font. 

For example,

<img width="384" alt="Screen Shot 2019-12-10 at 11 44 18 PM" src="https://user-images.githubusercontent.com/1478421/70549943-9914ff80-1ba7-11ea-8a11-a8ab4eb7a49e.png">

It looks fine to me, but it's no longer WordPress style. 

I'll remove it and add a note in the readme, but let me know if you want something else.

limonte commented on wordpress-admin/wordpress-admin.scss:
```diff
@@ -0,0 +1,192 @@
+}
+
+.swal2-title {
+  margin: 0;
```
use `$swal2-title-margin`

limonte commented on wordpress-admin/wordpress-admin.scss:
```diff
@@ -0,0 +1,192 @@
+.swal2-icon {
+  height: 32px;
+  transform: scale(0.7) translateY(24px);
+  animation: none !important;
```
use `$swal2-icon-animations`

limonte commented on wordpress-admin/package.json:
```diff
@@ -0,0 +1,23 @@
+    "*.css",
+    "*.scss"
+  ],
+  "main": "wordpress.css",
```
`wordpress-admin.css`

limonte commented on wordpress-admin/package.json:
```diff
@@ -0,0 +1,23 @@
+  "version": "3.0.7",
+  "repository": "sweetalert2/sweetalert2-themes",
+  "homepage": "https://sweetalert2.github.io/",
+  "description": "Wordpress admin theme for SweetAlert2",
```
`Wordpress` -> `WordPress`

limonte commented on wordpress-admin/README.md:
````diff
@@ -0,0 +1,77 @@
+To more closely match the WordPress style toast messages, use the following to get started:
+
+```js
+var notify = Swal.mixin({
````
`var` -> `const`

limonte commented on package.json:
```diff
@@ -5,24 +5,25 @@
+    "gulp-clean-css": "^4.2.0",
     "gulp-header": "^2.0.9",
-    "gulp-rename": "^1.2.2",
+    "gulp-rename": "^2.0.0",
```
reverse this line please, there's something wrong with `gulp-rename@2.0.0`

limonte commented on package.json:
```diff
@@ -5,24 +5,25 @@
   "homepage": "https://sweetalert2.github.io/",
```
you made changes in `package.json` please bump `yarn.lock` as well (`yarn install`), thanks!

KevinBatdorf commented:
Thanks @limonte Let me know if there's anything else.

limonte commented:
:tada: This PR is included in version 3.1.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@sweetalert2/themes)
- [GitHub release](https://github.com/sweetalert2/sweetalert2-themes/releases/tag/v3.1.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
